### PR TITLE
#9 - Bug fix

### LINKED ISSUE #9: Add mysql connection reset on error
The way it is now, the `Cannot enque query after fatal error` message pops-out after an error, blocking all subsequent queries from execution.

### PR DESCRIPTION
**Pull request by Fezzzi**
Closes #9

### CODE REVIEW

File: code/server/src/models/connection.ts [hunk 1]
```diff
@@ -11,6 +14,34 @@ const config: object = {
...
 };
 
-export const connection = mysql.createConnection(config);
+export class Connection {
+  private static __connection: mysql.Connection|null = null;
+
+  private static __createConnection = () => {
+    const connection = mysql.createConnection(config);
+    connection.on('error', err => {
+      if (err && err.fatal) {
```
Comment (sauermar) [inline code/server/src/models/connection.ts:23]: Create standalone error handler in connection.ts.

File: code/server/src/models/database.ts [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+import { Connection } from './connection';
 import { Logger } from '../helpers/logger';
 import { DB_LOG } from '../constants/logs';
 
 export const database = {
   query: (sql: string, cb: queryCallback) => {
-    connection.query(sql, (err: MysqlError | null, results?: any, fields?: FieldInfo[]) => {
+    Connection.get().query(sql, (err: MysqlError | null, results?: any, fields?: FieldInfo[]) => {
       Logger(DB_LOG, `${sql}; (${(err && err.message) || 'OK'})\n`);
+      if (err && err.fatal) {
```
Comment (sauermar) [inline code/server/src/models/database.ts:11]: Use the error handler from connection.ts.

File: code/server/src/models/connection.ts (hunk 1 above)
Comment (Fezzzi) [reply code/server/src/models/connection.ts]: Fixed [here](https://github.com/Fezzzi/Household-Chores/pull/10/commits/df91c2a3b76cb0477266a4db46e1397cbb108b9f#diff-f509ebf1c6486e48524aac270ff642b7)

File: code/server/src/models/database.ts (hunk 2 above)
Comment (Fezzzi) [reply code/server/src/models/database.ts]: Fixed [here](https://github.com/Fezzzi/Household-Chores/pull/10/commits/df91c2a3b76cb0477266a4db46e1397cbb108b9f#diff-4acfd404455a1add895ffa6ef4bd1dec)

**Pull request merged**